_posts/2022-12-16-aave-rebasing.md

```diff
@@ -0,0 +1,332 @@
+* Ben's rebased balance at this point would be 95.23 * 1.1025 == $105
+* so he'd have 5% more underlying than me, as expected (since the pool as a whole earned 5% while he had his money in it)
+
+And that's it! That is how interest works in Aave.
+
+### Flexible Voting
+
+[Flexible voting](https://www.scopelift.co/blog/introducing-flexible-voting) (or _flex voting_ for short)
+is a novel extension to Compound-style governance systems.
+It allows delegates to split their voting weight across For/Against/Abstain options, rather than having to put all of it behind one.
+
+If flex voting could be added to aTokens, this would unlock a large amount of
+value for Aave.
+
+To see why, imagine you hold UNI.
+UNI gets its intrinsic value from being the voting token for [Uniswap governance proposals](https://app.uniswap.org/#/vote).
+Anyone who holds UNI can delegate and vote on these proposals and help set the course of one of the biggest projects in crypto.
+
+So the main value of UNI comes from its use in voting.
+But it would be really nice if you could also deposit your UNI to earn yield at the same time.
+
+The trouble is that if you were to deposit UNI into a DeFi protocol (like Aave) you would lose your voting power.
+This is because the address that holds the tokens controls their voting weight.
+And if you deposit your UNI, the receiving contract -- not you -- would hold the tokens.
+
+So governance token holders are forced to choose: either participate in governance or earn yield.
+They cannot currently have both.
+
+If flex voting were added to aTokens and the Aave DAO, however, they could.
+
+In such a world, aToken holders would express their desired voting preference to
+the aToken contract, which would then cast its vote in the appropriate ratios to
+the governance contract.
+
+So, for example, if 35% of aUNI holders expressed a "No" voting preference on a
+given proposal, 60% expressed "Yes", and 5% "Abstain", then the aToken contract
+would cast its vote to the UNI governance system with 35% of its weight as
+Against, 60% of its weight as For, and 5% as Abstain.
+
+aUNI holders would effectively get to vote without having a UNI balance!
+
+While flex voting makes this dream scenario possible, it's not without
+technical hurdles.
+The next section will get into what these are and how we solved them.
+
+### Implementing a Flex Voting aToken
+
+If something like the dream scenario were to exist, it would have to take
+account of the fact that aToken balances are constantly increasing, and doing so
+at an ever-changing rate.
+
+After all, if I could have burned my aTokens and withdrawn a certain weight of
+voting tokens at the time of a proposal, any vote cast on my behalf by the aToken
+should reflect that weight.
+
+Let's look at a concrete example:
+
+* Ben deposits 100 UNI into Aave
+* a year passes and aUNI yields 50% interest (it was a _very_ good year)
+* I deposit 100 UNI
```

Comment:
Nit, maybe pick another name here and avoid "I" / first-person throughout? I don't think we've used "I" in the posts before. But we have used "we" and "our", e.g. in regards "our umbra supporters"

Reply:
Sure, how about "Matt"? :)

Reply:
Heh fine with me. Or something like charlie and dennis since we all have some relation to philly haha 😛